<?xml version="1.0"?>
<robot
    xmlns:xacro="http://www.ros.org/wiki/xacro" name="dragon_head" >

  <xacro:macro name="pr2_gripper" params="self"> 

    <joint name="${self}_left_gripper_joint" type="revolute">
      <axis xyz="0 0 1"/>
      <limit effort="1000.0" lower="0.0" upper="0.548" velocity="0.5"/>
      <origin rpy="0 0 0" xyz="0.015 0.01 0"/>
      <parent link="${self}_rear_junction"/>
      <child link="${self}_left_gripper"/>
    </joint>

    <link name="${self}_left_gripper">
      <visual>
        <origin rpy="0.0 0 0" xyz="0 0 0"/>
        <geometry>
          <mesh filename="package://aerial_robot_model/urdf/mesh_model/l_finger.dae"/>
        </geometry>
        <xacro:link_color />
      </visual>
    </link>

    <joint name="${self}_left_tip_joint" type="fixed">
      <parent link="${self}_left_gripper"/>
      <child link="${self}_left_tip"/>
    </joint>

    <link name="${self}_left_tip">
      <visual>
        <origin rpy="0.0 0 0" xyz="0.09137 0.00495 0"/>
        <geometry>
          <mesh filename="package://pr2_description/meshes/gripper_v0/l_finger_tip.dae"/>
        </geometry>
        <xacro:link_color />
      </visual>
    </link>

    <joint name="${self}_right_gripper_joint" type="revolute">
      <axis xyz="0 0 -1"/>
      <limit effort="1000.0" lower="0.0" upper="0.548" velocity="0.5"/>
      <origin rpy="0 0 0" xyz="0.015 -0.01 0"/>
      <parent link="${self}_rear_junction"/>
      <child link="${self}_right_gripper"/>
    </joint>

    <link name="${self}_right_gripper">
      <visual>
        <origin rpy="-3.1415 0 0" xyz="0 0 0"/>
        <geometry>
          <mesh filename="package://aerial_robot_model/urdf/mesh_model/l_finger.dae"/>
        </geometry>
        <xacro:link_color />
      </visual>
    </link>

    <joint name="${self}_right_tip_joint" type="fixed">
      <parent link="${self}_right_gripper"/>
      <child link="${self}_right_tip"/>
    </joint>

    <link name="${self}_right_tip">
      <visual>
        <origin rpy="-3.1415 0 0" xyz="0.09137 0.00495 0"/>
        <geometry>
          <mesh filename="package://pr2_description/meshes/gripper_v0/l_finger_tip.dae"/>
        </geometry>
        <xacro:link_color />
      </visual>
    </link>
  </xacro:macro>

  <xacro:macro name="dragon_head" params="self"> 
    <xacro:pr2_gripper self="${self}"/>
    <link name="${self}_rear_junction">
      <visual>
        <geometry>
          <box size="0.01 ${tail_width} ${abdomen_height}"/>
        </geometry>
        <xacro:link_color />
      </visual>
    </link>

  </xacro:macro>

  <xacro:macro name="dragon_head2" params="parent self"> 
    <xacro:pr2_gripper self="${self}"/>
    <joint name="${parent}_${self}_joint" type="fixed">
      <parent link="${parent}_rear_junction"/>
      <child link="${self}_rear_junction"/>
      <origin rpy="-1.5708  3.1415 0" xyz="-0.01 0 0"/>
    </joint>


    <link name="${self}_rear_junction">
      <visual>
        <geometry>
          <box size="0.01 ${tail_width} ${abdomen_height}"/>
        </geometry>
        <xacro:link_color />
      </visual>
    </link>

  </xacro:macro>

</robot>
